File: src/backend/Common/ExprCalc.Entities/MetadataParams/CalculationFilters.cs
using ExprCalc.Entities.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExprCalc.Entities.MetadataParams
{
    public record class CalculationFilters
    {
        public static CalculationFilters NoFilters { get; } = new CalculationFilters();

        public Guid? Id { get; init; }

        public string? CreatedBy { get; init; }

        public DateTime? CreatedAtMin { get; init; }
        public DateTime? CreatedAtMax { get; init; }

        public DateTime? UpdatedAtMin { get; init; }
        public DateTime? UpdatedAtMax { get; init; }

        public CalculationState? State { get; init; }

        public string? Expression { get; init; }

        public double? CalculationResultMin { get; init; }
        public double? CalculationResultMax { get; init; }
    }
}

File: src/frontend/app/models/calculationFilters.ts
import { DateTime, Uuid } from "../common";
import { CalculationState } from "./calculation";

export type CalculationFilters = {
    id?: Uuid;
    createdBy?: string;

    createdAtMin?: DateTime;
    createdAtMax?: DateTime;

    updatedAtMin?: DateTime;
    updatedAtMax?: DateTime;

    state?: CalculationState;
    expression?: string;
}